addons/account_audit_trail/models/account_invoice.py

```diff
@@ -0,0 +1,36 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import logging
+
+from odoo import api, models, _
+from odoo.exceptions import UserError
+
+_logger = logging.getLogger(__name__)
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    @api.ondelete(at_uninstall=False)
+    def _unlink_account_audit_trail_except_once_post(self):
+        if any(m.posted_before and m.company_id.check_account_audit_trail for m in self) and not self._context.get('force_delete'):
+            raise UserError(_("To keep the audit trail, you can not delete journal entries once they have been posted.\nInstead, you can cancel the journal entry."))
+
+    def unlink(self):
+        # Add logger here because in api ondelete account.move.line is deleted and we can't get total amount
+        logger_msg = False
+        if any(m.posted_before and m.company_id.check_account_audit_trail for m in self):
+            if self._context.get('force_delete'):
+                moves_details = ", ".join("{entry_number} ({move_id}) amount {amount_total} {currency} and partner {partner_name}".format(
+                    entry_number=m.name,
+                    move_id=m.id,
+                    amount_total=m.amount_total,
+                    currency=m.currency_id.name,
+                    partner_name=m.partner_id.display_name)
+                    for m in self)
```

Comment:
Be consistent with the opening and closing of brackets. They need to be closed on the same indentation level as they were opened to ease the reading of the blocks.

```suggestion
                moves_details = ", ".join("{entry_number} ({move_id}) amount {amount_total} {currency} and partner {partner_name}".format(
                    entry_number=move.name,
                    move_id=move.id,
                    amount_total=move.amount_total,
                    currency=move.currency_id.name,
                    partner_name=move.partner_id.display_name,
                ) for move in self)
```

Reply:
Shouldn't we give the detail of the lines, especially the balance per accounts?

Reply:
it's good to have.
message is like
```
Force deleted Journal Entries by Mitchell Admin (2)
Entries
INV/2023/00003 (27) amount 118 INR and partner Deco Addict
- Creditors (12) with balance -118 INR
- Product sale (13) with balance 100 INR
- Tax Payable (45) with balance 18 INR
```